### ISSUE TITLE
Handler keeps on waiting for a timeout even though the client has been disconnected.

### ISSUE THREAD

**Issue by alex-ant** [jcuga/golongpoll#4]
Won't it be better for the handler function to stop waiting for a timeout if the client has been disconnected?

Steps to reproduce the issue:

1. Create a new manager and register the handler:
  
  ```go
  manager, _ = golongpoll.CreateCustomManager(120, 1, true)
  http.HandleFunc("/events", func(w http.ResponseWriter, r *http.Request) {
    manager.SubscriptionHandler(w, r)
    fmt.Println("client disconnected")
  })
  ```
2. Connect to the server from a browser (it will start waiting for the response)
3. Manager's log: Adding Client (Category: "abc112" Client: c6650762-f3d7-4876-410c-c6df1ea1dfc0)
4. Without publishing anything disconnect the client (close browser's window or something)
5. The SubscriptionHandler is still running

I guess it'd be logical to terminate the handler's execution at once after the disconnect event.

**Comment by jcuga** [comment]
The SubscriptionHandler already listens for client disconnects and unregisters the subscription the moment a client disconnects regardless of whether or not any events occurred.  

However, from taking a second look at the code, I think you are right that the handler remains running.  I added the logic to unsubscribe, but it looks like the handler itself isn't told to exit.  I will try and fix this soon.

**Comment by jcuga** [comment]
Issue confirmed.  Though the fix is a little more complicated than I thought.  I will make time to fix this the right way sometime within the next few days.

**Comment by jcuga** [comment]
This should now be fixed via aba297b270a53663a6a5ef97d744aaf6f6d06104

I added a unit test to guard against this bug in the future.  If this looks good I can go ahead and close this.  

By fixing this, I actually simplified the previous code that fixes Issue #1, and identified another bug introduced in some prior commits where client subscriptions were never being cleaned up on the manager side.  So it's good you pointed this out.  Thanks!
